### PR TITLE
Update cli image docker client to 17.06

### PR DESCRIPTION
**Pull request by jakolehm**
Docker 17.06 supports multi-stage builds.

Related issue: #3092

### CODE REVIEW

File: cli/Dockerfile [hunk 1]
```diff
@@ -1,12 +1,16 @@
-FROM alpine:3.5
+FROM alpine:3.6
 MAINTAINER Kontena, Inc. <info@kontena.io>
 
-ADD https://get.docker.com/builds/Linux/x86_64/docker-1.10.3 /usr/local/bin/docker
+ADD https://download.docker.com/linux/static/stable/x86_64/docker-17.06.2-ce.tgz /tmp
```
Comment (jnummelin) [inline cli/Dockerfile]: This way the tgz package will be "left behind" in the layers forever. The later `rm -rf /tmp/docker*` just masks it out. Why not add it in the `RUN` with `wget` or something?

**Comment by SpComb** [comment]
What happens if you run the `kontena/cli` image on a machine with an older version of Docker? I'm not entirely sure how a newer CLI version behaves against an older API version.

Not that it's a problem, it just means that we may require Docker 17.06+ when using the `kontena stack build` stuff in the CLI image.

**Comment by jnummelin** [comment]
> I'm not entirely sure how a newer CLI version behaves against an older API version.

One can control the API used with `DOCKER_API_VERSION` env variable.

File: cli/Dockerfile (hunk 1 above)
Comment (jakolehm) [reply cli/Dockerfile]: Fixed.

**Comment by jakolehm** [comment]
@jnummelin @SpComb PTAL

File: cli/Dockerfile [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
 MAINTAINER Kontena, Inc. <info@kontena.io>
 
-ADD https://get.docker.com/builds/Linux/x86_64/docker-1.10.3 /usr/local/bin/docker
 ARG CLI_VERSION
 
 RUN apk update && \
   apk --update add ruby ruby-json ruby-bigdecimal ruby-io-console \
-  ca-certificates libssl1.0 openssl libstdc++ && \
+  curl ca-certificates libssl1.0 openssl libstdc++ && \
+  curl -sL https://download.docker.com/linux/static/stable/x86_64/docker-17.06.2-ce.tgz > /tmp/docker.tgz && \
```
Comment (SpComb) [inline cli/Dockerfile]: Could be simplified to just `curl ... | tar xz`, leaving the `/tmp/docker.tgz` out.

**Comment by SpComb** [comment]
Testing against the v1.4 branch, this works with Docker 17.09 on the host machine:

```
tehobari:~/kontena/kontena/cli :: bundle exec rake release:build_docker && docker run -v /var/run/docker.sock:/var/run/docker.sock --entrypoint=docker kontena/cli:1.4.3 version
docker rmi kontena/cli:1.4.3 || true
Untagged: kontena/cli:1.4.3
docker build --build-arg CLI_VERSION=1.4.3 --no-cache --pull -t kontena/cli:1.4.3 .
Sending build context to Docker daemon  1.561MB
Step 1/8 : FROM alpine:3.6
3.6: Pulling from library/alpine
Digest: sha256:42c18571af45ffc440bd5127de9ad604f1683e4799dbf7bf588aaee8f95c18f1
Status: Image is up to date for alpine:3.6
 ---> e2cd449cde75
Step 2/8 : MAINTAINER Kontena, Inc. <info@kontena.io>
 ---> Running in 07c9c0e11287
 ---> f99c74e5d569
Removing intermediate container 07c9c0e11287
Step 3/8 : ARG CLI_VERSION
 ---> Running in a5688b8dee9c
 ---> dab389b304a2
Removing intermediate container a5688b8dee9c
Step 4/8 : RUN apk update &&   apk --update add ruby ruby-json ruby-bigdecimal ruby-io-console   curl ca-certificates libssl1.0 openssl libstdc++ &&   curl -sL https://download.docker.com/linux/static/stable/x86_64/docker-17.06.2-ce.tgz > /tmp/docker.tgz &&   cd /tmp &&   tar zxf docker.tgz &&   mv docker/docker /usr/local/bin/ &&   rm -rf /tmp/docker* &&   chmod +x /usr/local/bin/docker
 ---> Running in 27f3e6de0028
fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/community/x86_64/APKINDEX.tar.gz
v3.6.2-233-gde42cedd77 [http://dl-cdn.alpinelinux.org/alpine/v3.6/main]
v3.6.2-234-ga6787bf312 [http://dl-cdn.alpinelinux.org/alpine/v3.6/community]
OK: 8437 distinct packages available
fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/community/x86_64/APKINDEX.tar.gz
(1/22) Installing ca-certificates (20161130-r2)
(2/22) Installing libssh2 (1.8.0-r1)
(3/22) Installing libcurl (7.57.0-r0)
(4/22) Installing curl (7.57.0-r0)
(5/22) Installing libcrypto1.0 (1.0.2n-r0)
(6/22) Installing libssl1.0 (1.0.2n-r0)
(7/22) Installing libgcc (6.3.0-r4)
(8/22) Installing libstdc++ (6.3.0-r4)
(9/22) Installing openssl (1.0.2n-r0)
(10/22) Installing libffi (3.2.1-r3)
(11/22) Installing gdbm (1.12-r0)
(12/22) Installing gmp (6.1.2-r0)
(13/22) Installing ncurses-terminfo-base (6.0_p20170930-r0)
(14/22) Installing ncurses-terminfo (6.0_p20170930-r0)
(15/22) Installing ncurses-libs (6.0_p20170930-r0)
(16/22) Installing readline (6.3.008-r5)
(17/22) Installing yaml (0.1.7-r0)
(18/22) Installing ruby-libs (2.4.3-r0)
(19/22) Installing ruby (2.4.3-r0)
(20/22) Installing ruby-bigdecimal (2.4.3-r0)
(21/22) Installing ruby-io-console (2.4.3-r0)
(22/22) Installing ruby-json (2.4.3-r0)
Executing busybox-1.26.2-r9.trigger
Executing ca-certificates-20161130-r2.trigger
OK: 31 MiB in 33 packages
 ---> f5820b663f0a
Removing intermediate container 27f3e6de0028
Step 5/8 : RUN apk --update add --virtual build-dependencies ruby-dev build-base openssl-dev &&   gem install kontena-cli --no-rdoc --no-ri -v ${CLI_VERSION} &&   apk del build-dependencies
 ---> Running in 3e339dbf4007
fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/community/x86_64/APKINDEX.tar.gz
(1/21) Installing libgmpxx (6.1.2-r0)
(2/21) Installing gmp-dev (6.1.2-r0)
(3/21) Installing pkgconf (1.3.7-r0)
(4/21) Installing ruby-dev (2.4.3-r0)
(5/21) Installing binutils-libs (2.28-r3)
(6/21) Installing binutils (2.28-r3)
(7/21) Installing isl (0.17.1-r0)
(8/21) Installing libgomp (6.3.0-r4)
(9/21) Installing libatomic (6.3.0-r4)
(10/21) Installing mpfr3 (3.1.5-r0)
(11/21) Installing mpc1 (1.0.3-r0)
(12/21) Installing gcc (6.3.0-r4)
(13/21) Installing musl-dev (1.1.16-r14)
(14/21) Installing libc-dev (0.7.1-r0)
(15/21) Installing g++ (6.3.0-r4)
(16/21) Installing make (4.2.1-r0)
(17/21) Installing fortify-headers (0.8-r0)
(18/21) Installing build-base (0.5-r0)
(19/21) Installing zlib-dev (1.2.11-r0)
(20/21) Installing openssl-dev (1.0.2n-r0)
(21/21) Installing build-dependencies (0)
Executing busybox-1.26.2-r9.trigger
OK: 204 MiB in 54 packages
Successfully installed excon-0.49.0
Successfully installed necromancer-0.4.0
Successfully installed equatable-0.5.0
Successfully installed tty-color-0.4.2
Successfully installed pastel-0.7.2
Successfully installed tty-cursor-0.5.0
Successfully installed wisper-2.0.0
Building native extensions.  This could take a while...
Successfully installed hitimes-1.2.6
Successfully installed timers-4.1.2
Successfully installed tty-prompt-0.13.1
Successfully installed clamp-1.1.2
Successfully installed ruby_dig-0.0.2
Successfully installed public_suffix-3.0.1
Successfully installed addressable-2.5.2
Successfully installed launchy-2.4.3
Successfully installed hash_validator-0.7.1
Successfully installed retriable-2.1.0
Successfully installed opto-1.8.7
Successfully installed semantic-1.6.0
Successfully installed liquid-4.0.0
Successfully installed tty-screen-0.6.4
Successfully installed unicode_utils-1.4.0
Successfully installed unicode-display_width-1.1.3
Successfully installed verse-0.5.0
Successfully installed tty-table-0.9.0
Successfully installed websocket-extensions-0.1.3
Building native extensions.  This could take a while...
Successfully installed websocket-driver-0.6.5
Successfully installed kontena-websocket-client-0.1.1
Successfully installed kontena-cli-1.4.3
29 gems installed
(1/21) Purging build-dependencies (0)
(2/21) Purging ruby-dev (2.4.3-r0)
(3/21) Purging gmp-dev (6.1.2-r0)
(4/21) Purging libgmpxx (6.1.2-r0)
(5/21) Purging build-base (0.5-r0)
(6/21) Purging g++ (6.3.0-r4)
(7/21) Purging gcc (6.3.0-r4)
(8/21) Purging binutils (2.28-r3)
(9/21) Purging libatomic (6.3.0-r4)
(10/21) Purging libgomp (6.3.0-r4)
(11/21) Purging make (4.2.1-r0)
(12/21) Purging libc-dev (0.7.1-r0)
(13/21) Purging musl-dev (1.1.16-r14)
(14/21) Purging fortify-headers (0.8-r0)
(15/21) Purging openssl-dev (1.0.2n-r0)
(16/21) Purging zlib-dev (1.2.11-r0)
(17/21) Purging pkgconf (1.3.7-r0)
(18/21) Purging binutils-libs (2.28-r3)
(19/21) Purging isl (0.17.1-r0)
(20/21) Purging mpc1 (1.0.3-r0)
(21/21) Purging mpfr3 (3.1.5-r0)
Executing busybox-1.26.2-r9.trigger
OK: 31 MiB in 33 packages
 ---> e3351e052c03
Removing intermediate container 3e339dbf4007
Step 6/8 : VOLUME /root
 ---> Running in 815a15719e83
 ---> e0ceaaeba63b
Removing intermediate container 815a15719e83
Step 7/8 : WORKDIR /root
 ---> 5e0dad07f68d
Removing intermediate container f738c69e087b
Step 8/8 : ENTRYPOINT /usr/bin/kontena
 ---> Running in 5b77ba2cbd7c
 ---> 07c5be6cfb8c
Removing intermediate container 5b77ba2cbd7c
Successfully built 07c5be6cfb8c
Successfully tagged kontena/cli:1.4.3
docker rmi kontena/cli:latest || true
Untagged: kontena/cli:latest
docker tag kontena/cli:1.4.3 kontena/cli:latest
docker rmi kontena/cli:1.4 || true
Error response from daemon: conflict: unable to remove repository reference "kontena/cli:1.4" (must force) - container 8282680e9458 is using its referenced image b692ff7b0fd0
docker tag kontena/cli:1.4.3 kontena/cli:1.4
Client:
 Version:      17.06.2-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   cec0b72
 Built:        Tue Sep  5 19:57:21 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.09.1-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   19e2cf6
 Built:        Thu Dec  7 22:23:00 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

Also tested `kontena stack build` to still work.

**Pull request merged**